### ISSUE TITLE
Documentation template tab order should match visual order

### ISSUE THREAD

**Issue by aduth** [uswds/uswds#4594]
**Is your feature request related to a problem? Please describe.**

#### As a developer
 Who uses USWDS in my project, I expect that my generated stylesheets only includes styles relevant for my site, so that I'm not shipping unnecessary code to my users, prolonging their load times.

#### As a user

As a user who navigates a site using USWDS, I expect that load times are reasonable, so that I can access the content I need in a timely manner.

As a user, [I expect that I can tab through content in a logical order](https://www.w3.org/TR/WCAG20-TECHS/H4.html), so that there is not a disconnect between the visual and mechanical presentation of the content.

**Describe the solution you'd like**

Remove the file `src/stylesheets/components/_layout.scss`, or more specifically the `.usa-layout-docs__sidenav` and `.usa-layout-docs__main` style selectors.

Adjust [`components/layouts/docs-inner.njk`](https://github.com/uswds/uswds/blob/develop/src/components/layouts/docs-inner.njk) such that either (a) the placement of the side navigation is consistently before or after the main content, regardless of device size, or (b) the markup is duplicated and shown/hidden at the appropriate viewport size, but in such a way that ensures the [DOM order is always consistent with the visual order](https://www.w3.org/TR/WCAG20-TECHS/C27.html).

**Describe alternatives you've considered**

If USWDS needs styling for its own documentation, this should be compiled separate from what's included in USWDS.

**Additional context**

Related issue on `order` concerns: #4424

**Comment by thisisdano** [comment]
A couple issues here. The first, about including the `usa-layout-docs` styles in the main stylesheet is perhaps less critical after USWDS 3.0 — which more easily allows teams to use only the components (or "packages") their project needs in their stylesheets. [More about that in the USWDS Packages docs](https://designsystem.digital.gov/components/packages/).

These files support the [Documentation page template](https://designsystem.digital.gov/templates/documentation-page/) so I think we should keep them available.

The bigger issue is about DOM order and visual order — and that's one we can can and should address.

**Comment by mejiaj** [comment]
For reference there are components that reorder elements:

- [Date picker](https://github.com/uswds/uswds/blob/aab7f87c537337aafd0d0867ed1311ed227f94f9/packages/usa-date-picker/src/styles/_usa-date-picker.scss#L374)
- [In page navigation](https://github.com/uswds/uswds/blob/aab7f87c537337aafd0d0867ed1311ed227f94f9/packages/usa-in-page-navigation/src/styles/_usa-in-page-navigation.scss#L25)
- [Nav](https://github.com/uswds/uswds/blob/aab7f87c537337aafd0d0867ed1311ed227f94f9/packages/usa-nav/src/styles/_usa-nav.scss#L133)
